### ISSUE TITLE
/run/pihole-FTL.port is empty

### ISSUE THREAD

**Issue by kapoorankush** [pi-hole/FTL#840]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*
- [X] The issue I am reporting isn't a duplicate 

**How familiar are you with the codebase?:**
2

**[BUG | ISSUE] Expected Behaviour:**
/run/pihole-FTL.port is expected to contain the port number on which FTL is listening

**[BUG | ISSUE] Actual Behaviour:**
/run/pihole-FTL.port is empty, but the file exists and was created at the timestamp when FTL was started. File is owned by pihole.

```
pi@pi1b2:~ $ ls -l /run/pihole-FTL.port
-rw-r--r-- 1 pihole pihole 0 Jul 17 08:25 /run/pihole-FTL.port
```

**[BUG | ISSUE] Steps to reproduce:**
Encountered issue on 3 different attempts to run pi-hole on Raspberry Pi Model B Rev 2. Issue manifested itself with FTL being reported as offline when running the chronometer with `pihole -c`

```
  Hostname: pi1b2              (Raspberry Pi 1, Model B)
    Uptime: 22:37:20
 Task Load: 0.45 0.19 0.12     (Active: 3 of 39 tasks)
 CPU usage: 18%                (700 MHz @ 325k)
 RAM usage: 17%                (Used: 74 MB of 432 MB)
 HDD usage: 4%                 (Used: 2 GB of 58 GB)
  LAN addr: 10.0.0.249         (Gateway: 10.0.0.1)
   Pi-hole: Active             (Blocking: 0 sites)
 Ads Today: 0%                 (Total: 0 of 0)
Local Qrys: 0%                 (4 DNS servers)
   Blocked: FTL offline
Top Advert:
Top Domain:
Top Client:                                                                                                                                                                                                                                  ^C
pi@pi1b2:~ $
```

FTL is working fine, but shows offline. 

Checked `/opt/pihole/chronometer.sh` to find 
`ftl_port=$(cat /run/pihole-FTL.port 2> /dev/null) `
getting set to null. 

As a test, overriding ftL_port to the 4711 in `/opt/pihole/chronometer.sh` resulted in normal (expected output.  

**Log file output [if available]**

pihole-FTL starts fine:

```
pi@pi1b2:~ $ sudo service pihole-FTL status
● pihole-FTL.service - LSB: pihole-FTL daemon
   Loaded: loaded (/etc/init.d/pihole-FTL; generated)
   Active: active (exited) since Fri 2020-07-17 08:25:04 PDT; 2h 3min ago
     Docs: man:systemd-sysv-generator(8)
   Memory: 0B
   CGroup: /system.slice/pihole-FTL.service

Jul 17 08:24:59 pi1b2 systemd[1]: Starting LSB: pihole-FTL daemon...
Jul 17 08:25:00 pi1b2 pihole-FTL[15543]: Not running
Jul 17 08:25:00 pi1b2 su[15567]: (to pihole) root on none
Jul 17 08:25:00 pi1b2 su[15567]: pam_unix(su:session): session opened for user pihole by (uid=0)
Jul 17 08:25:04 pi1b2 pihole-FTL[15543]: FTL started!
Jul 17 08:25:04 pi1b2 systemd[1]: Started LSB: pihole-FTL daemon.
```

Output of `/var/log/pihole-FTL.log`
```
[2020-07-17 08:25:02.655 15598M] Using log file /var/log/pihole-FTL.log
[2020-07-17 08:25:02.657 15598M] ########## FTL started! ##########
[2020-07-17 08:25:02.659 15598M] FTL branch: master
[2020-07-17 08:25:02.660 15598M] FTL version: v5.1
[2020-07-17 08:25:02.662 15598M] FTL commit: b522646
[2020-07-17 08:25:02.664 15598M] FTL date: 2020-07-15 22:26:33 +0100
[2020-07-17 08:25:02.665 15598M] FTL user: pihole
[2020-07-17 08:25:02.667 15598M] Compiled for arm (compiled on CI) using arm-linux-gnueabihf-gcc (crosstool-NG crosstool-ng-1.22.0-88-g8460611) 4.9.3
[2020-07-17 08:25:02.668 15598M] Starting config file parsing (/etc/pihole/pihole-FTL.conf)
[2020-07-17 08:25:02.670 15598M]    SOCKET_LISTENING: only local
[2020-07-17 08:25:02.672 15598M]    AAAA_QUERY_ANALYSIS: Show AAAA queries
[2020-07-17 08:25:02.674 15598M]    MAXDBDAYS: max age for stored queries is 365 days
[2020-07-17 08:25:02.676 15598M]    RESOLVE_IPV6: Resolve IPv6 addresses
[2020-07-17 08:25:02.677 15598M]    RESOLVE_IPV4: Resolve IPv4 addresses
[2020-07-17 08:25:02.679 15598M]    DBINTERVAL: saving to DB file every minute
[2020-07-17 08:25:02.680 15598M]    DBFILE: Using /etc/pihole/pihole-FTL.db
[2020-07-17 08:25:02.682 15598M]    MAXLOGAGE: Importing up to 24.0 hours of log data
[2020-07-17 08:25:02.684 15598M]    PRIVACYLEVEL: Set to 0
[2020-07-17 08:25:02.686 15598M]    IGNORE_LOCALHOST: Show queries from localhost
[2020-07-17 08:25:02.688 15598M]    BLOCKINGMODE: Null IPs for blocked domains
[2020-07-17 08:25:02.690 15598M]    ANALYZE_ONLY_A_AND_AAAA: Disabled. Analyzing all queries
[2020-07-17 08:25:02.692 15598M]    DBIMPORT: Importing history from database
[2020-07-17 08:25:02.692 15598M]    PIDFILE: Using /run/pihole-FTL.pid
[2020-07-17 08:25:02.694 15598M]    SOCKETFILE: Using /run/pihole/FTL.sock
[2020-07-17 08:25:02.695 15598M]    SETUPVARSFILE: Using /etc/pihole/setupVars.conf
[2020-07-17 08:25:02.697 15598M]    MACVENDORDB: Using /etc/pihole/macvendor.db
[2020-07-17 08:25:02.699 15598M]    GRAVITYDB: Using /etc/pihole/gravity.db
[2020-07-17 08:25:02.701 15598M]    PARSE_ARP_CACHE: Active
[2020-07-17 08:25:02.702 15598M]    CNAME_DEEP_INSPECT: Active
[2020-07-17 08:25:02.703 15598M]    DELAY_STARTUP: No delay requested.
[2020-07-17 08:25:02.705 15598M]    NICE: Set process niceness to -10 (default)
[2020-07-17 08:25:02.707 15598M]    BLOCK_ESNI: Enabled, blocking _esni.{blocked domain}
[2020-07-17 08:25:02.708 15598M]    NAMES_FROM_NETDB: Enabled, trying to get names from network database
[2020-07-17 08:25:02.710 15598M] Finished config file parsing
[2020-07-17 08:25:02.722 15598M] Database version is 6
[2020-07-17 08:25:02.723 15598M] Database successfully initialized
[2020-07-17 08:25:02.730 15598M] New upstream server: 2606:4700:4700::1001 (0/1024)
[2020-07-17 08:25:02.735 15598M] New upstream server: 127.0.0.1 (1/1024)
[2020-07-17 08:25:02.757 15598M] New upstream server: 2606:4700:4700::1111 (2/1024)
[2020-07-17 08:25:02.800 15598M] New upstream server: 1.0.0.1 (3/1024)
[2020-07-17 08:25:02.842 15598M] Resizing "/FTL-queries" from 229376 to 458752
[2020-07-17 08:25:02.883 15598M] Resizing "/FTL-strings" from 4096 to 8192
[2020-07-17 08:25:02.900 15598M] New upstream server: 1.1.1.1 (4/1024)
[2020-07-17 08:25:02.962 15598M] Resizing "/FTL-queries" from 458752 to 688128
[2020-07-17 08:25:03.073 15598M] Resizing "/FTL-queries" from 688128 to 917504
[2020-07-17 08:25:03.194 15598M] Resizing "/FTL-strings" from 8192 to 12288
[2020-07-17 08:25:03.205 15598M] Resizing "/FTL-queries" from 917504 to 1146880
[2020-07-17 08:25:03.320 15598M] Resizing "/FTL-strings" from 12288 to 16384
[2020-07-17 08:25:03.373 15598M] Resizing "/FTL-queries" from 1146880 to 1376256
[2020-07-17 08:25:03.531 15598M] Resizing "/FTL-queries" from 1376256 to 1605632
[2020-07-17 08:25:03.682 15598M] Resizing "/FTL-queries" from 1605632 to 1835008
[2020-07-17 08:25:03.729 15598M] Resizing "/FTL-strings" from 16384 to 20480
[2020-07-17 08:25:03.988 15598M] Resizing "/FTL-queries" from 1835008 to 2064384
[2020-07-17 08:25:04.196 15598M] Resizing "/FTL-queries" from 2064384 to 2293760
[2020-07-17 08:25:04.217 15598M] Resizing "/FTL-strings" from 20480 to 24576
[2020-07-17 08:25:04.417 15598M] Imported 40711 queries from the long-term database
[2020-07-17 08:25:04.422 15598M]  -> Total DNS queries: 40711
[2020-07-17 08:25:04.428 15598M]  -> Cached DNS queries: 20893
[2020-07-17 08:25:04.429 15598M]  -> Forwarded DNS queries: 12563
[2020-07-17 08:25:04.431 15598M]  -> Blocked DNS queries: 7255
[2020-07-17 08:25:04.433 15598M]  -> Unknown DNS queries: 0
[2020-07-17 08:25:04.437 15598M]  -> Unique domains: 762
[2020-07-17 08:25:04.438 15598M]  -> Unique clients: 7
[2020-07-17 08:25:04.440 15598M]  -> Known forward destinations: 5
[2020-07-17 08:25:04.442 15598M] Successfully accessed setupVars.conf
[2020-07-17 08:25:04.514 15605M] PID of FTL process: 15605
[2020-07-17 08:25:04.523 15605M] Reloading DNS cache
[2020-07-17 08:25:04.551 15605/T15607] Listening on port 4711 for incoming IPv6 telnet connections
[2020-07-17 08:25:04.552 15605/T15608] Listening on Unix socket
[2020-07-17 08:25:04.554 15605/T15606] Listening on port 4711 for incoming IPv4 telnet connections
[2020-07-17 08:25:04.558 15605M] Blocking status is enabled
[2020-07-17 08:25:04.661 15605M] INFO: No regex blacklist entries found
[2020-07-17 08:25:04.678 15605M] INFO: No regex whitelist entries found
[2020-07-17 08:25:04.751 15605M] Compiled 0 whitelist and 0 blacklist regex filters for 7 clients in 91.6 msec
```

**Device specifics**

Hardware Type: Raspberry Pi 1 Model B Rev 2
OS: Raspbian

```
pi@pi1b2:~ $ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
pi@pi1b2:~ $ uname -a
Linux pi1b2 4.19.118+ #1311 Mon Apr 27 14:16:15 BST 2020 armv6l GNU/Linux
pi@pi1b2:~ $  netstat -anp | grep 4711
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
tcp        0      0 127.0.0.1:4711          0.0.0.0:*               LISTEN      -
tcp6       0      0 ::1:4711                :::*                    LISTEN      -
pi@pi1b2:~ $
```

**Comment by yubiuser** [comment]
See here: 
https://github.com/pi-hole/FTL/pull/838

**Comment by DL6ER** [comment]
This should be fixed by the release of Pi-hole FTL 5.2. Feel free to re-open of create a new ticket with new information if the issue persists.
